ntbs-service/Services/HospitalSearchService.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Castle.Core.Internal;
+using ntbs_service.DataAccess;
+using ntbs_service.Models.ReferenceEntities;
+
+namespace ntbs_service.Services
+{
+    public interface IHospitalSearchService
+    {
+        Task<IList<Hospital>> OrderQueryableAsync(string searchKeyword);
+    }
+
+    class HospitalSearchService : IHospitalSearchService
+    {
+        private readonly IReferenceDataRepository _referenceDataRepository;
+
+        public HospitalSearchService(IReferenceDataRepository referenceDataRepository)
+        {
+            _referenceDataRepository = referenceDataRepository;
+        }
+
+        public async Task<IList<Hospital>> OrderQueryableAsync(string searchKeyword)
+        {
+            var searchKeywords = searchKeyword.Split(" ")
+                .Where(x => !x.IsNullOrEmpty())
+                .Select(s => s.ToLower()).ToList();
```

Comment:
These lines are duplicated between all of the search services. I think it would be better to do this parsing in the calling class, and pass a list of keywords to each service.

Also, rather than excluding empty entries directly, I think there's an overload of Split where you can specify an option called something like `RemoveEmptyEntries`